plugins/call_button/call_button.tsx

```diff
@@ -22,14 +27,19 @@ type Props = {
     channelMember?: ChannelMembership;
     pluginCallComponents: PluginComponent[];
     sidebarOpen: boolean;
+    currentUserId: string;
     customButton?: JSX.Element;
+    userId?: string;
+    channelToStartCall?: Channel | null;
+    startCallInDM?: boolean;
 }
 
-export default function CallButton({pluginCallComponents, currentChannel, channelMember, sidebarOpen, customButton}: Props) {
+export default function CallButton({pluginCallComponents, currentChannel, channelMember, sidebarOpen, customButton, channelToStartCall, currentUserId, userId, startCallInDM = false}: Props) {
```

Comment:
This seems a touch overly complex. I'd expect to pass a channel ID where the call should be started in and not much more as anything else should be likely dealt with by the caller.

Maybe we could even re-purpose the `currentChannel` arg for this without breaking anything. What do you think?

Reply:
@streamer45  This is exactly what I was thinking initially. I thought I just need to pass channel to start a call. Then I realized it wasnt that easy. 

IMO re-purposing `currentChannel` is possible where current user has already a DM channel with the other user. However, this is not always the case. And if there is no DM channel exists, I need to create DM Channel only if user clicks the call button, not on component mount. If I create DM channel in `index` file, an unnecessary new DM channel is created everytime when user opens profile popover

That is why implemented logic like this:
- pass `channelToStartCall` i.e. already existing DM Channel
- if `channelToStartCall` doesn't exist, create a DM Channel using `currentUserId` and `userId` which are passed as prop on user click call button

Reply:
Thanks for the added context @sinansonmez . 

I feel most of the complication is coming from the fact that we are slightly abusing the plugin button. Maybe the simplest solution would be to stop relying on that but implement the logic directly in the profile popover. 

After all you could always access the callback needed to join a call from `state.plugins.components.CallButton[0].action` (with the appropriate undefined checks of course) and use that in a click handler of your custom button.

I understand this proposal would mean reverting most of the work done so far but it honestly feels cleaner to me. That said, happy to wait to hear @AshishDhama's thoughts given his superior experience on this side of the app.

Reply:
@streamer45 That is an interesting (also ref. Zef's Platformer 😄 ) to point of view. 

No worries about reverting the changes. For me the important thing is to impletement it in a best possible way 👨‍💻 

I would wait for Ashish's feedback and try to make it more cleaner

Reply:
@streamer45 I agree we should move this logic out of the `call_button.tsx`, but in my opinion, profile popover isn’t the right place either. I would vote for a custom button or hook handling this logic instead of putting it in the `call_button.tsx` or `profile_popover.tsx`. cc// @sinansonmez 
@streamer45 From the logic in `call_button.tsx.`, we expect multiple call plugins, right? If we are expecting a multiple-call plugin I think we will be rewriting the button UI, In that case, I think some modifications in `call_button.tsx` and a custom hook would be best.
But if in the `profile popover` we just want our calls plugin button I would say let’s create a custom button component and use that. @laneycs could you please guide us here?

Reply:
TL;DR Move the logic to create DM, out of `call_button.tsx` to either hook or custom button.

Reply:
@laneycs please let me know what do you think about 2 potential options mentioned by Ashish

Reply:
> From the logic in call_button.tsx., we expect multiple call plugins, right?

Correct, the button is meant to support multiple plugins (e.g. Zoom, Jitsi) with Calls being the default one. That said I really don't know if the same is expected from the popover. My guess would be no as we also trying to integrate Calls more into the main product and at some point it will likely be part of the suite.

To unblock this, since it's a bug fix, I propose to extract some of the logic as @AshishDhama suggested and create a custom button component. @sinansonmez Let me know if that makes sense.

Reply:
Thanks @streamer45 this makes sense. I reverted the previous changes in `call_button.tsx` and I created a custom component for `ProfilePopover`